src/sherlock/nginx_tailer.py

```diff
@@ -33,7 +33,18 @@ async def tail_nginx_log() -> None:
             )
 
             while True:
-                line = await resp.content.readline()
+                # apparently readline() times out after six minutes
+                # ...so you'd think you could just catch that error and
+                # retry, but then you get intermittent pod-not-ready
+                # errors
+                # ...so let's try a slightly smaller controlled timeout
+                # and see if that helps any.
+                #
+                # Spoiler: it did.
+                try:
+                    line = await asyncio.wait_for(resp.content.readline(), 300)
+                except asyncio.TimeoutError:
+                    continue
```

Comment:
I don't think this will work... the continue will rerun line 35 and we'll just keep trying to read from probably a dead response? It seems like that should maybe be a break so that it will make a new query up on line 21. But I think that's what it does without this, since we're already in a try.

What error were you hitting about the pod not ready? I'm very confused about this, what pod wasn't ready?

I have seen the timeout happen but when it does it just normally picks up from where it left off, so I'm not sure if you're seeing something else?

Reply:
You don't happen to have the logs for the error do you?  (I know it's been a while)

Reply:
Well, I never put the fix anywhere but data-dev.  It was an error in the GKE dashboard (I'm working on reducing meaningless error spam).  I'll see if I can go pull it out from data-int.

Reply:
I don't think it's reading from a dead response, and I think we have probably only ever seen it in dev.  It's actually (I think) legitimate: if there is nothing to tail for more than six minutes, readline() times out and that throws an error.  I don't know why just trapping that error causes GKE to freak out.

But if readline() hasn't yielded anything in five minutes, and we kill and rerun the readline(), then that particular problem never occurs.  We still don't get any data, but we're trying to.

My guess is that dev is the only place where, legitimately, *nothing at all happens* for more than six minutes at a time.

Reply:
I can replicate it, I'm pretty sure, by reverting dev to the baseline sherlock configuration and killing all the monkeys.  I'm working on dev right now but I'll try to get to it when I'm off working somewhere else, so we can generate the errors.

Reply:
I've done that.  I'll capture some errors.